test/fixtures/dialects/snowflake/snowflake_copy_into_table.yml

```diff
@@ -270,7 +270,7 @@ file:
               from_expression_element:
                 table_expression:
                   table_reference:
-                    stage_path: '@ext.stage/path1'
+                    identifier: '@ext.stage/path1'
```

Comment:
I remember having loads of issues trying to get a file path recognised when I first looked at this months ago. Think I even gave up on it and left it for later. Was a while ago so can’t remember all the details but just warning you can end up going down a rabbit while with this. So maybe worth leaving as is for now and tackling later if needs be.

Reply:
What are we doing for other dialects? Or is this the only dialect that does this so far?

Multi-part references have been a challenge already, although in a different way, for BigQuery, where references can be indefinitely long, e.g. `a.b.c.d.e.f.g`, so the rules sometimes can't tell which part is the database, dataset, or column.

Similar to some of the recent templating work, I wonder if we may be better off keeping the schema simple and instead providing Python functions (potentially dialect specific) for processing this stuff. It's a tricky design question. On the one hand, the parse tree can be more explicit, but every time we add or change something, we risk breaking other areas (and/or adding complexity).

See #2278 for a recent example of this.

If you search `src/sqlfluff/rules` for the word "references", you'll see several rules that currently deal with references. It may be worth adding a new test(s) to each of those rules.

Reply:
Following up on @barrywhart's test suggestion these are the potentially impacted rules:
![image](https://user-images.githubusercontent.com/80432516/149664916-b342098d-cbe7-4666-a7eb-3ab9107f38e1.png)

L054 isn't relevant.
L057 should be investigated for the more strict configs since a stage_path would likely contain some special characters.
L044 you've already done.

L020, L025, L026, L027, L028, L031 are all worth investigating as I feel like they could apply to stages as well, so would be worth investigating to see what's happening with them atm

Reply:
(if we check these and nothing is broken but we think there are enhancements to be done we can leave those for a later PR. It's just to check nothing is broken atm)

Reply:
>just warning you can end up going down a rabbit while with this

I agree with this. In _theory_, parsing it into pieces sounds like a cleaner, more proper approach. I worry about rabbit holes as well, and I'm okay with deferring a tricky decision like this until it matters. Often, I find that deferring a decision like this makes it easier -- by the time we have to decide, we may have more context to help us decide (more rules, more dialects with similar needs, etc.).

Reply:
> Following up on @barrywhart's test suggestion these are the potentially impacted rules: ![image](https://user-images.githubusercontent.com/80432516/149664916-b342098d-cbe7-4666-a7eb-3ab9107f38e1.png)
> 
> L054 isn't relevant. L057 should be investigated for the more strict configs since a stage_path would likely contain some special characters. L044 you've already done.
> 
> L020, L025, L026, L027, L028, L031 are all worth investigating as I feel like they could apply to stages as well, so would be worth investigating to see what's happening with them atm

@jpy-git I ran `sqlfluff lint` on the test case locally, which makes me think we are fine on all these rules. Is your ask to add a Snowflake stage test for each of them?

Reply:
Well there's a difference between the test case triggering fatal errors on the rules vs checking if the rules behave as you would expect and if the change could introduce more subtle bugs.

That being said we can merge this in as it fixes the linked issue and I'll create an issue around investigating if the other rules behave as expected for stages.